near-sdk/src/collections/vec/mod.rs

```diff
@@ -0,0 +1,258 @@
+//! A vector implemented on a trie. Unlike standard vector does not support insertion and removal
+//! of an element results in the last element being placed in the empty position.
+// TODO update these docs
+
+mod impls;
+mod iter;
+
+use self::iter::{Iter, IterMut};
+use crate::collections::append_slice;
+use crate::{env, CacheCell, CacheEntry, EntryState, IntoStorageKey};
+use borsh::{BorshDeserialize, BorshSerialize};
+use std::collections::{btree_map::Entry, BTreeMap};
+use std::ptr::NonNull;
+
+const ERR_INCONSISTENT_STATE: &[u8] = b"The collection is an inconsistent state. Did previous smart contract execution terminate unexpectedly?";
```

Comment:
std::env::panic should take `str` rather than `u8` argument, the same issue as with log, and withe the same compat concerns.

Reply:
Yeah I agree, I can make an additional API for that and the log under the unstable flag and group it with #366 or open a separate PR for that. This change is a bit more annoying to put under a flag as the interactions don't go through a macro like `log!`